usefulUtilities/smartBoard/scripts/domainEntities/smartBoard/boardButtonClient.js

```diff
@@ -0,0 +1,77 @@
+//
+//  boardButtonClient.js
+//
+//  Additional code by Milad Nazeri 10/30/2019
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+(function(){
+// ENTITY SIGNALS
+    // Grab the current color, set the button type, get the zone ID, and send a buttonLoaded signal to the zone
+    function preload(entityID){
+        _this.entityID = entityID;
+        var props = Entities.getEntityProperties(entityID, ["name", "position", "color"]);
+        _this.entityName = props.name;
+        _this.color = props.color
+        var entityPosition = props.position;
+        if (_this.entityName === "Screenshare_whiteboard_button") {
+            _this.buttonType = "whiteboard";
+        } else {
+            _this.buttonType = "screenshare"
+        }
+        _this.screenshareZone = Entities.findEntitiesByName("Screenshare_Zone", entityPosition, 0.5)[0];
```

Comment:
Please use parent/child relationships to find entities instead of finding entities spatially.

I'm only going to put this comment on the PR once, but it goes for the rest of the code too, in case there are any more instances of code like this :)